src/DynamoCore/Library/FunctionDescriptor.cs

```diff
@@ -89,7 +89,7 @@ public FunctionDescriptor(FunctionDescriptorParams funcDescParams)
                 {
                     x.Function = this;
                     return x;
-                });
+                }).ToList();
```

Comment:
Did you do this based on the recommendation from ReSharper that says *possible multi enumeration*?